fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -183,6 +185,44 @@ public DatatableData retrieveDatatable(final String datatable) {
         return datatableData;
     }
 
+    @Override
+    public List<JsonObject> queryDataTable(String datatable, String columnFilter, String valueFilter, String resultColumns) {
+        Arrays.asList(datatable, columnFilter, valueFilter, resultColumns).forEach(SQLInjectionValidator::validateDynamicQuery);
+        if(columnFilter == null || valueFilter == null) {
+            return new ArrayList<>(0);
+        }
+
+        List<String> resultColumnNames = resultColumns == null ? Collections.emptyList() : Stream.of(resultColumns.split(",")).toList();
+        final String sql = "select " + (resultColumns == null ? "" : resultColumns.trim()) + " from " + datatable + " where "
```

Comment:
Please validate the provided filter column and resultColumns by the datatable metadata informations.
e.g.: this.genericDataService.fillResultsetColumnHeaders(dataTableName); -> it will give you the column headers metadata

Reply:
changing the code accordingly